staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -44,6 +44,15 @@ const (
 	// resourceVersionTooHighRetrySeconds is the seconds before a operation should be retried by the client
 	// after receiving a 'too high resource version' error.
 	resourceVersionTooHighRetrySeconds = 1
+
+	// eventRecordDuration is time duration of events we want to keep.
+	eventFreshDuration = 5 * time.Minute
+
+	// defaultLowerBoundCapacity is a default value for event cache capacity's lower bound.
+	defaultLowerBoundCapacity = 100
+
+	// defaultLowerBoundCapacity is a default value for event cache capacity's upper bound.
+	defaultUpperBoundCapacity = 100 * 1024
```

Comment:
That is lower than 150k that I initially proposed. And it's already clear we want be able to keep 5 minutes of history of leases in 5k-node cluster.That is probably fine, but we should mention it explicitly in the comment.

Reply:
Done

Reply:
This isn't fully true now, because 102400 is not enough to keep 5min of leases in 5k-node cluster.
Let's change to sth like:
```
// defaultUpperBoundCapacity  should be able to keep eventFreshDuration of history.
// With the current 102400 value though, it's not not enough for leases in 5k-node cluster, but that is conscious decision.
// TODO: Validate if the current value is high enough for large scale clusters.

Reply:
Done.